### ISSUE TITLE
Support NeMo MegatronCLIPModel

### ISSUE THREAD

**Issue by athitten** [Lightning-AI/lightning-thunder#337]
### 🚀 Feature

NeMo's MegatronCLIPModel is a vision-language foundation model.

Initial `examine`:

`Found 40 distinct operations, of which 29 (72.5%) are supported`

### Work items
- [ ] [Implement `TensorBase.type` #177 ](https://github.com/Lightning-AI/lightning-thunder/issues/177)
- [ ] [`torch.nn.functional.normalize` #334 ](https://github.com/Lightning-AI/lightning-thunder/issues/334)
- [ ] [`TensorBase.masked_fill_` #335](https://github.com/Lightning-AI/lightning-thunder/issues/335)
- [ ] [`TensorBase.expand_as` #330 ](https://github.com/Lightning-AI/lightning-thunder/issues/330)
- [ ] [`TensorBase.view_as` #328 ](https://github.com/Lightning-AI/lightning-thunder/issues/328)
- [ ] [`_VaribaleFunctionsClass.baddbmm` of torch #326 ](https://github.com/Lightning-AI/lightning-thunder/issues/326)
- [ ] [`TensorBase.all` #336](https://github.com/Lightning-AI/lightning-thunder/issues/336)
- [ ] [`_VaribaleFunctionsClass.empty` of torch #331 ](https://github.com/Lightning-AI/lightning-thunder/issues/331)
- [ ] [`TensorBase.is_cuda` #329 ](https://github.com/Lightning-AI/lightning-thunder/issues/329)
- [ ] [`_set_grad_enabled` of torch._C #332 ](https://github.com/Lightning-AI/lightning-thunder/issues/332)
- [ ] [`device` of torch #327 ](https://github.com/Lightning-AI/lightning-thunder/issues/327)

cc @tfogal
